src/components/Reports/Subscribers/Subscribers.js

```diff
@@ -0,0 +1,108 @@
+import React, { useState, useEffect } from 'react';
+import { InjectAppServices } from '../../../services/pure-di';
+import { useParams } from 'react-router-dom';
+import { Loading } from '../../Loading/Loading';
+import SafeRedirect from '../../SafeRedirect';
+import SubscriberGdprPermissions from '../GDPRPermissions/SubscriberGdprPermissions';
+import SubscriberSentCampaigns from '../SubscriberSentCampaigns/SubscriberSentCampaigns';
+import { useIntl } from 'react-intl';
+import { Helmet } from 'react-helmet';
+import SubscriberInfo from '../../shared/SubscriberInfo/SubscriberInfo';
+
+const sections = {
+  history: {
+    Component: SubscriberSentCampaigns,
+    titleKey: 'subscriber_history.title',
+    descriptionKey: 'subscriber_history.description',
+  },
+  gdpr: {
+    Component: SubscriberGdprPermissions,
+    titleKey: 'subscriber_gdpr.header_title',
+    descriptionKey: 'subscriber_gdpr.header_description',
+  },
+};
+
+const Subscribers = ({ dependencies: { dopplerApiClient } }) => {
+  const { email, section } = useParams();
+  const [state, setState] = useState({ loading: true });
+  const intl = useIntl();
+  const _ = (id, values) => intl.formatMessage({ id: id }, values);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setState({ loading: true });
+      const responseSubscriber = await dopplerApiClient.getSubscriber(email);
+      if (responseSubscriber.success) {
+        const subscriber = {
+          ...responseSubscriber.value,
+          // TODO: move this to api client
+          firstName: responseSubscriber.value.fields.find((x) => x.name === 'FIRSTNAME'),
+          lastName: responseSubscriber.value.fields.find((x) => x.name === 'LASTNAME'),
+        };
+
+        setState({
+          subscriber: subscriber,
+          loading: false,
+        });
+      } else {
+        setState({ redirect: true });
+      }
+    };
+    fetchData();
+  }, [dopplerApiClient, email]);
+
+  if (!sections[section] || state.redirect) {
+    return <SafeRedirect to="/Lists/MasterSubscriber/" />;
```

Comment:
What happens if you have two different errors: wronh section and wrong subscriber? ie. `/subscribers/noexisto@mail.com/seccionincorrecta`?

Reply:
It should redirect. At least, this is what is doing.

Reply:
I know, but there is a kind of race condition, right?

What is more, what happen if the section name is wrong and the email is fine? ie. `/subscribers/right@email.com/seccionincorrecta`?

Reply:
I've added a test for that.

Reply:
for the race condition? mmmm

Reply:
https://github.com/FromDoppler/doppler-webapp/blob/54e6fb1b1f60157b57dfbe95e14e0627234d910e/src/components/Reports/Subscribers/Subscribers.test.js#L108

Reply:
Si, pero no estás testeando lo que está pasando, por ejemplo, el ajax se hace igual?

Reply:
Si, el ajax se hace igual, aunque no debería si el section esta mal. Quizas habría que hacer el ajax solo si el section existe 🤔 si no, no tiene sentido.

Reply:
mmm, por ahí simplemente poner un TODO por ahora.